src/deserialize.zig

```diff
@@ -154,6 +154,11 @@ pub fn deserialize(comptime T: type, serialized: []const u8, out: *T, allocator:
 
                 return end;
             },
+            .One => {
+                out.* = try allocator.create(ptr.child);
+                return deserialize(ptr.child, serialized, out.*, allocator);
```

Comment:
note that the rlp size/marker aren't consumed since RLP doesn't know about pointers. It's a simple recursion.